iml-agent/src/action_plugins/stratagem/server.rs

```diff
@@ -377,7 +384,12 @@ pub fn stream_fidlists(
             })
     });
 
-    future::join_all(mailbox_files).map(|_| ())
+    future::join_all(mailbox_files)
+        .and_then(|_| {
+            remove_dir(parent_dir);
```

Comment:
This needs to be returned from the `and_then` combinator, otherwise it won't get polled to completion.

Something like:

```rust
        .and_then(|_| {
            remove_dir(parent_dir).from_err()
         })
```